### PR TITLE
Detecting when installing over MingW's Git Bash, fixing paths

### PR DESCRIPTION
**Pull request by jeffque**
## Description

Detecting when the instalelr is being run over MingW, so that it works correctly with git-bash

I'm noting this as `WIP` because @Hixie told me to

>  yes please update documentation as part of any fix, so that doxs are always accurate

So I'll go doc hunting later

## Related Issues

#8860

@goderbauer have wrote a list of things to tackle, of which those have been solved:

- [x] update_dart_sdk.sh needs to be taught the download location for the Windows version of Dart (easy)
- [x] flutter needs to call pub.bat, not pub (easy)
- [x] Pub needs to properly work in Git bash (dart-lang/sdk#29102)
- [ ] During hot runs, keystrokes need to be honored

This last one was not tackled, as hitting `<enter>` is annoying but it still works.

## Tests

I have not added any tests, as I have no idea on how to write test for this MingW detector bash script.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

About the **I updated/added** check: there was no relevant doc around the bash code, so I have not added anything about this... So I'm leaving this unchecked.

About the **All [...] tests are passing** check: all the `flutter` commands seem to be working. So, is there any other test that should be done?

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by jeffque** [comment]
@zanderso , I think that you should be the one reviewing this

**Comment by jeffque** [comment]
Removing the WIP:

I assumed that, in the main repo, there was the doc about it. I assumed, and hunted for, the source for [this page](https://flutter.dev/docs/get-started/install/windows). Hunted but to no avail, there wasn't any line about needing the PowerShell. Also, could not undestand the issues with the key binding to hotreload, as in MingW's terminal won't work, but VSCode's terminal bounded to git-bash executable work.

@christopherfujino , I think that this iteration is, for now, complete. At least it allows people to do flutter stuff over git-bash without the need to work around.

**Comment by jeffque** [comment]
Oh, and sorry for the long wait and no relevant change to the code base...

File: bin/flutter [hunk 1]
```diff
@@ -150,6 +150,13 @@ DART_SDK_PATH="$FLUTTER_ROOT/bin/cache/dart-sdk"
 DART="$DART_SDK_PATH/bin/dart"
 PUB="$DART_SDK_PATH/bin/pub"
 
+case "$(uname -s)" in
```
Comment (zanderso) [inline bin/flutter]: Please add some comments explaining why this is here.

Comment (jeffque) [reply bin/flutter]: Maybe something like this?

```suggestion
# If running over git-bash, overrides the default UNIX
# executables with win32 executables
case "$(uname -s)" in
```

Comment (zanderso) [reply bin/flutter]: Yup, that sounds good. Thanks.

**Comment by jeffque** [comment]
@zanderso @christopherfujino , I think it is ready again to be reviewed

**Review by zanderso — APPROVED** [review]

**Comment by tvolkert** [comment]
It looks like this was merged on red without an explanation as to why.

**Comment by jonahwilliams** [comment]
Build looked green when we merged it in triage.

**Pull request merged**